toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -247,22 +326,28 @@ func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string,
 	)
 	defer func() {
 		if err != nil {
-			span.RecordError(fmt.Errorf("image customization failed"))
-			span.SetStatus(codes.Error, "image customization failed")
+			errorName := "Unset" // default
+			if namedErrors := GetAllImageCustomizerErrors(err); len(namedErrors) > 0 {
+				errorName = namedErrors[len(namedErrors)-1].Name()
+			}
+			span.SetAttributes(
+				attribute.String("error.name", errorName),
```

Comment:
Use `StringSlice` here to log all the error names?

Reply:
that's true, we can log all of them. Updated